src/orion/algo/hyperband.py

```diff
@@ -145,18 +170,33 @@ class Hyperband(BaseAlgorithm):
 
     """
 
-    def __init__(self, space, seed=None, repetitions=numpy.inf):
-        self.brackets = []
-        super(Hyperband, self).__init__(space, seed=seed, repetitions=repetitions)
-
-        self.trial_to_brackets = {}  # Stores Point id (with no fidelity) -> Bracket
-        self._trials_info = {}
+    def __init__(
+        self,
+        space: Space,
+        seed: int | Sequence[int] | None = None,
+        repetitions: int | float = numpy.inf,
+    ):
+        # NOTE: Need to set the attribute before calling super().__init__ because it calls seed_rng.
+        self.brackets: Optional[list[HyperbandBracket]] = []
+        super().__init__(space, repetitions=repetitions, seed=seed)
+        self.seed = seed
+        # Stores Point id (with no fidelity) -> Bracket (int)
+        self.trial_to_brackets: dict[str, int] = {}
 
-        fidelity_index = self.fidelity_index
+        fidelity_index: str | None = self.fidelity_index
         if fidelity_index is None:
             raise RuntimeError(SPACE_ERROR)
 
-        fidelity_dim = space[fidelity_index]
+        fidelity_dim: Fidelity = space[fidelity_index]
+
+        # NOTE: This isn't a Fidelity, it's a TransformedDimension<Fidelity>
+        from orion.core.worker.transformer import TransformedDimension
+
+        # TODO: Double-check with @bouthilx: Currently bypassing (possibly more than one)
+        # `TransformedDimension` wrappers to get the 'low', 'high' and 'base' attributes.
+        while isinstance(fidelity_dim, TransformedDimension):
```

Comment:
Perhaps we should try to avoid wrapping the Fidelity dimension in the transformed space. This fix is hacky but necessary until this issue is resolved. In any case, I think it should be done in another PR when we have time. I'd be in favor of leaving this here as-is for this PR.

Reply:
Can I mark this as resolved?

Reply:
Oh nevermind I'll remove the TODO first.

Reply:
Fixed in 1b74ab37bfcb95f3741d52a875b91d495c0db89c
(might not be pushed yet)

Reply:
Create issue #856